qt/scientific_interfaces/ISISReflectometry/GUI/Preview/PreviewPresenter.cpp

```diff
@@ -204,7 +217,25 @@ void PreviewPresenter::notifyRegionChanged() {
   m_dockedWidgets->setRectangularROIState(false);
   m_dockedWidgets->setEditROIState(true);
 
-  runReduction();
+  if (isRegionSelectionChanged()) {
+    m_regionSelector->deselectAllSelectors();
+    runReduction();
+  }
+}
+
+namespace {
+bool isRegionChanged(std::optional<std::vector<double>> modelValue, std::vector<double> viewValue) {
+  return !(!modelValue.has_value() && viewValue.empty()) && modelValue != viewValue;
+}
+} // namespace
+
+bool PreviewPresenter::isRegionSelectionChanged() {
+  return isRegionChanged(m_model->getSelectedRegion(ROIType::Signal),
+                         m_regionSelector->getRegion(roiTypeToString(ROIType::Signal))) ||
+         isRegionChanged(m_model->getSelectedRegion(ROIType::Background),
+                         m_regionSelector->getRegion(roiTypeToString(ROIType::Background))) ||
+         isRegionChanged(m_model->getSelectedRegion(ROIType::Transmission),
+                         m_regionSelector->getRegion(roiTypeToString(ROIType::Transmission)));
```

Comment:
Just a suggestion on style in case you agree - I was wondering if the below alternative might help cut down some of the duplication in `PreviewPresenter::isRegionSelectionChanged`, although at the expense of a bit more boilerplate (I haven't checked the below, so apologies if it doesn't work!):

```
bool PreviewPresenter::isRegionSelectionChanged() {
  return isRegionChanged(ROIType::Signal) ||
         isRegionChanged(ROIType::Background) ||
         isRegionChanged(ROIType::Transmission);
}

bool PreviewPresenter::isRegionChanged(ROIType roiType) {
  auto oldValue = m_model->getSelectedRegion(roiType);
  auto newValue = m_regionSelector->getRegion(roiTypeToString(roiType));
  return !(!oldValue.has_value() && newValue.empty()) && oldValue != newValue;
}
```

Also, I'm trying to decide, is `!(!modelValue.has_value() && viewValue.empty())` equivalent to `(modelValue.has_value() || !viewValue.empty())` - i.e. either the old or new ROI has to have a value?